app/src/full/java/io/homeassistant/companion/android/settings/SettingsWearViewModel.kt

```diff
@@ -0,0 +1,117 @@
+import com.google.android.gms.wearable.Wearable
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import kotlinx.coroutines.launch
+
+class SettingsWearViewModel : ViewModel() {
+
+    private lateinit var integrationUseCase: IntegrationRepository
+
+    private val TAG = "SettingsWearViewModel"
+    private val CAPABILITY_WEAR_FAVORITES = "send_home_favorites"
+
+    var entities = mutableStateMapOf<String, Entity<*>>()
+        private set
+    var favoriteEntityIds = mutableStateListOf<String>()
+
+    fun init(integrationUseCase: IntegrationRepository) {
+        this.integrationUseCase = integrationUseCase
+        loadEntities()
+    }
+
+    private fun loadEntities() {
+        viewModelScope.launch {
+            integrationUseCase.getEntities().forEach {
+                entities[it.entityId] = it
+            }
+        }
+    }
+
+    fun saveHomeFavorites(data: String, item: DataItem) {
+        getFavorites(DataMapItem.fromDataItem(item).dataMap)
+        favoriteEntityIds.clear()
+        favoriteEntityIds.addAll(
+            data.removeSurrounding("[", "]").split(", ").map { it }
+        )
+    }
+
+    fun onEntitySelected(checked: Boolean, entityId: String, activity: Activity) {
+        if (checked)
+            favoriteEntityIds.add(entityId)
+        else
+            favoriteEntityIds.remove(entityId)
+        sendHomeFavorites(favoriteEntityIds.toList(), activity)
+    }
+
+    private fun sendHomeFavorites(favoritesList: List<String>, activity: Activity) = viewModelScope.launch {
+        Log.d(TAG, "sendHomeFavorites")
+
+        val putDataRequest = PutDataMapRequest.create("/save_home_favorites").run {
+            dataMap.putString("favorites", favoritesList.toString())
+            setUrgent()
+            asPutDataRequest()
+        }
+
+        Wearable.getDataClient(activity).putDataItem(putDataRequest).apply {
+            addOnSuccessListener { Log.d(TAG, "Successfully sent favorites to wear") }
+            addOnFailureListener { Log.d(TAG, "Failed to send favorites to wear") }
+        }
+    }
+
+    fun findExistingFavorites(activity: Activity) {
+        Log.d(TAG, "Finding existing favorites")
+        Tasks.await(Wearable.getDataClient(activity).getDataItems(Uri.parse("wear://*/home_favorites"))).apply {
+            Log.d(TAG, "Found existing favorites: ${this.count}")
+            this.forEach {
+                val data = getFavorites(DataMapItem.fromDataItem(this.first()).dataMap)
+                Log.d(TAG, "Favorites: $data")
+                favoriteEntityIds.clear()
+                favoriteEntityIds.addAll(
+                    data.removeSurrounding("[", "]").split(", ").map { it }
+                )
+            }
+        }
+    }
+
+    fun getFavorites(map: DataMap): String {
+        map.apply {
+            return getString("favorites", "")
+        }

```

Comment:
This feels strange as is.
```suggestion
        return map.getString("favorites", "")
```

Reply:
thank you :)